src/methylbert/data/finetune_data_generate.py

```diff
@@ -138,36 +145,41 @@ def _get_methylseq(dmr, bam_file_path: str, k: int, methyl_caller: str):
 
     if ncores > 1:
         with mp.Pool(ncores) as pool:
-            # Convert read sequences to k-mer sequences 
-            seqs = pool.map(partial(_get_methylseq, 
+            # Convert read sequences to k-mer sequences
+            seqs = pool.map(partial(_get_methylseq,
                                     bam_file_path = bam_file_path, k=k, methyl_caller=methyl_caller),
                             dmrs.to_dict("records"))
     else:
-        seqs = [_get_methylseq(dmr, bam_file_path = bam_file_path, k=k, methyl_caller = methyl_caller) for dmr in dmrs.to_dict("records")]
-        
+        seqs = [_get_methylseq(dmr, bam_file_path = bam_file_path, k=k,
+                               methyl_caller = methyl_caller)
+                for dmr in dmrs.to_dict("records")]
+
     # Filter None values that means no overlapping read with the given DMR
     seqs = list(filter(lambda i: i is not None, seqs))
     if len(seqs) > 0:
         return pd.concat(seqs, ignore_index=True)
     else:
         warnings.warn(f"Zero reads were extracted from {bam_file_path}")
+        return pd.DataFrame([])
 
 def finetune_data_generate(
         f_dmr: str,
         output_dir: str,
         f_ref: str,
         sc_dataset: str = None,
-        input_file: str = None, 
+        input_file: str = None,
         n_mers: int = 3,
-        split_ratio: float = 1.0,
+        split_ratios: List[float] = [0.8, 0.1, 0.1],
         n_dmrs: int = -1,
         n_cores: int = 1,
         seed: int = 950410,
         ignore_sex_chromo: bool = True,
-        methyl_caller: str = "bismark"
+        methyl_caller: str = "bismark",
+        verbose: int = 2,
+        read_extract_sequences_func: Optional[callable] = None
```

Comment:
Is this an option for your own function?

Reply:
Exactly! I find it more user friendly this way, one can have a custom read extraction different from the original one without the need to fork the package